flake.nix

```diff
@@ -3,396 +3,110 @@
-          '';
-
-          configureFlags = [
-            # See https://github.com/ghdl/ghdl/pull/2058
-            "--disable-werror"
-            "--enable-synth"
-          ] ++ lib.optionals (backend == "llvm") [
-            "--with-llvm-config=${llvm.dev}/bin/llvm-config"
-          ];
-
-          hardeningDisable = [ "format" ];
-
-          enableParallelBuilding = true;
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
+    in {
 
-          passthru = {
-            # run with either of
-            # nix-build -A ghdl-mcode.passthru.tests
-            # nix-build -A ghdl-llvm.passthru.tests
-            tests = {
-              simple = callPackage ./test-simple.nix { inherit backend; };
+      # Provide some binary packages for selected system types.
+      packages = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+          inherit (pkgs) lib callPackage stdenv fetchgit fetchFromGitHub;
+        in rec {
+          default = yosys;
+
+          ghdl = pkgs.ghdl;
+
+          abc-verifier = pkgs.abc-verifier.overrideAttrs (_: rec {
+            version = "yosys-0.17";
+            src = fetchFromGitHub {
+              owner = "yosyshq";
+              repo = "abc";
+              rev = "09a7e6dac739133a927ae7064d319068ab927f90" # == version
+              ;
+              hash = "sha256-+1UcYjK2mvhlTHl6lVCcj5q+1D8RUTquHaajSl5NuJg=";
             };
-          };
+            passthru.rev = src.rev;
+          });
 
-          meta = with lib; {
-            homepage = "https://github.com/ghdl/ghdl";
-            description = "VHDL 2008/93/87 simulator";
-            maintainers = with maintainers; [ lucus16 thoughtpolice ];
-            platforms = platforms.linux;
-            license = licenses.gpl2;
-          };
-        };
+          yosys-ghdl = pkgs.yosys-ghdl;
 
-        abc-verifier = with final; stdenv.mkDerivation rec {
-          pname   = "abc-verifier";
-          version = "for-yosys-0.17";
+          # FIXME(jl): we're not sure why this yosys derivaiton is defined here,
+          # - is it the required version of yosys to use with nextpnr-xilinx?
+          yosys = (pkgs.yosys.overrideAttrs (prev: rec {
+            version = "0.17";
 
-          src = fetchFromGitHub {
-            owner = "yosyshq";
-            repo  = "abc";
-            rev   = "09a7e6d";
-            hash  = "sha256-+1UcYjK2mvhlTHl6lVCcj5q+1D8RUTquHaajSl5NuJg=";
-          };
-
-          nativeBuildInputs = [ cmake ];
-          buildInputs = [ readline ];
+            src = fetchFromGitHub {
+              owner = "yosyshq";
+              repo = "yosys";
+              rev = "${prev.pname}-${version}";
+              hash = "sha256-IjT+G3figWe32tTkIzv/RFjy0GBaNaZMQ1GA8mRHkio=";
+            };
 
-          installPhase = "mkdir -p $out/bin && mv abc $out/bin";
+            doCheck = true; # FIXME(ac): can we turn these back on?
```

Comment:
Yes, I think so. Those tests just took a long time and hindered the development process, so I turned them off.